qiskit_aer/backends/plugin/aer_backend_plugin.py

```diff
@@ -0,0 +1,116 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2018, 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Aer simulator backend transpiler plug-in
+"""
+from qiskit.transpiler.preset_passmanagers.plugin import PassManagerStagePlugin
+from qiskit.transpiler import PassManager, TransformationPass
+from qiskit.transpiler.passes import BasisTranslator
+from qiskit.transpiler.passes import UnitarySynthesis
+from qiskit.transpiler.passes import HighLevelSynthesis
+from qiskit.circuit.equivalence_library import SessionEquivalenceLibrary as sel
+from qiskit.circuit.measure import Measure
+from qiskit.circuit import ControlFlowOp
+from qiskit.converters import circuit_to_dag
+from qiskit_aer.backends.name_mapping import NAME_MAPPING
+
+
+class AerBackendRebuildGateSetsFromCircuit(TransformationPass):
```

Comment:
I think this should actually be an `AnalysisPass` you're not actually transforming the circuit, but are updating the target instead.

Reply:
Because there are only translation and scheduling plugins, I used translation plugin to analyze before going to optimization pass